puppetmaster/applications/__init__.py

```diff
@@ -0,0 +1,26 @@
+import logging
+
+from abc import ABC
+
+from puppetmaster.events import Events, Event
+from puppetmaster.providers import BaseProvider
+
+
+class BaseApplication(ABC):
+    """
+    Base Application class.
+
+    Inherit for shared methods.
+    """
+
+    def __init__(self):
+        raise NotImplementedError("Must use a real application.")
```

Comment:
It would be better to decorate these and `pass`, in case someone accidentally uses `super()`:

```suggestion
    @abc.abstractmethod
    def __init__(self):
        pass
```

Reply:
Also, I don't think that you need to explicitly define an abstract `__init__` since the ABC base should prevent direct instantiation anyway.